packages/editor-ui/src/views/CredentialsView.vue

```diff
@@ -125,7 +127,7 @@ listenForModalChanges({
 });
 
 const onFilter = (resource: Resource, newFilters: BaseFilters, matches: boolean): boolean => {
-	const Resource = resource as ICredentialsResponse & { needsSetup: boolean };
+	const Resource = resource as CredentialsResource;
```

Comment:
Could this be more typesafe? Like adding the typecheck above here `isCredentialsResource`?